jetty-core/jetty-io/src/main/java/org/eclipse/jetty/io/RetainableByteBuffer.java

```diff
@@ -1411,30 +1439,19 @@ public DynamicCapacity(ByteBufferPool pool, boolean direct, long maxSize, int ag
          */
         public DynamicCapacity(ByteBufferPool pool, boolean direct, long maxSize, int aggregationSize, int minRetainSize)
         {
-            this(new ArrayList<>(), pool, direct, maxSize, aggregationSize, minRetainSize);
+            this(null, new ByteBufferPool.Sized(pool, direct, maxSize > 0 && maxSize < 8192L ? (int)maxSize : aggregationSize), maxSize, minRetainSize);
         }
 
-        private DynamicCapacity(List<RetainableByteBuffer> buffers, ByteBufferPool pool, boolean direct, long maxSize, int aggregationSize, int minRetainSize)
+        private DynamicCapacity(List<RetainableByteBuffer> buffers, ByteBufferPool.Sized pool, long maxSize, int minRetainSize)
         {
             super();
-            _pool = pool == null ? ByteBufferPool.NON_POOLING : pool;
-            _direct = direct;
+            _pool = pool == null ? ByteBufferPool.SIZED_NON_POOLING : pool;
             _maxSize = maxSize < 0 ? Long.MAX_VALUE : maxSize;
-            _buffers = buffers;
+            _buffers = buffers == null ? new ArrayList<>() : buffers;
 
-            if (aggregationSize < 0)
-            {
-                _aggregationSize = (int)Math.min(_maxSize, 8192L);
-            }
-            else
-            {
-                if (aggregationSize > _maxSize)
-                    throw new IllegalArgumentException("aggregationSize(%d) must be <= maxCapacity(%d)".formatted(aggregationSize, _maxSize));
-                _aggregationSize = aggregationSize;
-            }
             _minRetainSize = minRetainSize;
 
-            if (_aggregationSize == 0 && _maxSize >= Integer.MAX_VALUE && _minRetainSize != 0)
+            if (_pool.getSize() == 0 && _maxSize >= Integer.MAX_VALUE && _minRetainSize != 0)
```

Comment:
This check can go as `ByteBufferPool.Sized` does not allow 0 size.

Reply:
It does now.  We need 0 size so that can indicate that aggregation is not to be done, but we still might need a pool for some copies.

Reply:
I think there is only one usage of this (see lines 2004-2006 in append and the javdoc of aggregationSize above).   Perhaps we don't need this mode, but we still want the ability to make a buffer that will only retain and never aggregate.

Reply:
Since the sized pool now allows 0 byte buffers, this check can stay.